website/server/controllers/api-v3/tasks/groups.js

```diff
@@ -0,0 +1,187 @@
+    if (reqValidationErrors) throw reqValidationErrors;
+
+    let user = res.locals.user;
+
+    let group = await Group.getGroup({user, groupId: req.params.groupId, populateLeader: false});
+    if (!group) throw new NotFound(res.t('groupNotFound'));
+
+    if (group.leader !== user._id) throw new NotAuthorized(res.t('onlyGroupLeaderCanEditTasks'));
+
+    let tasks = await createTasks(req, res, {user, undefined, group});
+
+    res.respond(201, tasks.length === 1 ? tasks[0] : tasks);
+
+    return null;
+  },
+};
+
+/**
+ * @api {get} /api/v3/tasks/group/:groupId Get a group's tasks
+ * @apiVersion 3.0.0
+ * @apiName GetGroupTasks
+ * @apiGroup Task
+ *
+ * @apiParam {UUID} groupId The id of the group from which to retrieve the tasks
+ * @apiParam {string="habits","dailys","todos","rewards"} type Optional query parameter to return just a type of tasks
+ *
+ * @apiSuccess {Array} data An array of tasks
+ */
+api.getGroupTasks = {
+  method: 'GET',
+  url: '/tasks/group/:groupId',
+  middlewares: [authWithHeaders()],
+  async handler (req, res) {
+    req.checkParams('groupId', res.t('groupIdRequired')).notEmpty().isUUID();
+    let types = Tasks.tasksTypes.map(type => `${type}s`);
+    req.checkQuery('type', res.t('invalidTaskType')).optional().isIn(types);
+
+    let validationErrors = req.validationErrors();
+    if (validationErrors) throw validationErrors;
+
+    let user = res.locals.user;
+
+    let group = await Group.getGroup({user, groupId: req.params.groupId, populateLeader: false});
+    if (!group) throw new NotFound(res.t('groupNotFound'));
+
+    let tasks = await getTasks(req, res, {user, group});
+    res.respond(200, tasks);
+  },
+};
+
+/**
+ * @api {post} /api/v3/tasks/:taskId/assign/:assignedUserId Assign a group task to a user
+ * @apiDescription Assigns a user to a group task
+ * @apiVersion 3.0.0
+ * @apiName AssignTask
+ * @apiGroup Task
+ *
+ * @apiParam {UUID} taskId The id of the task that will be assigned
+ * @apiParam {UUID} userId The id of the user that will be assigned to the task
+ *
+ * @apiSuccess data An object if a single task was created, otherwise an array of tasks
+ */
+api.assignTask = {
+  method: 'POST',
+  url: '/tasks/:taskId/assign/:assignedUserId',
+  middlewares: [authWithHeaders()],
+  async handler (req, res) {
+    req.checkParams('taskId', res.t('taskIdRequired')).notEmpty().isUUID();
+    req.checkParams('assignedUserId', res.t('userIdRequired')).notEmpty().isUUID();
+
+    let reqValidationErrors = req.validationErrors();
+    if (reqValidationErrors) throw reqValidationErrors;
+
+    let user = res.locals.user;
+    let assignedUserId = req.params.assignedUserId;
+    let assignedUser = await User.findById(assignedUserId);
+
+    let taskId = req.params.taskId;
+    let task = await Tasks.Task.findByIdOrAlias(taskId, user._id);
+
+    if (!task) {
+      throw new NotFound(res.t('taskNotFound'));
+    }
+
+    if (!task.group || !task.group.id) {
```

Comment:
checking for task.group should not be necessary as it'll always exist, even if empty